gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/storage/AWSStorageService.java

```
@@ -4,87 +4,117 @@
+                                                               .bucket(bucketName)
+                                                               .key(filename)
+                                                               .build();
+        s3Client.deleteObject(deleteRequest);
+        LOGGER.info("{} was successfully deleted", filename);
     }
 
     public String getFileLocation(String filename) {
-        // https://docs.aws.amazon.com/AmazonS3/latest/userguide/ShareObjectPreSignedURL.html
-        // sharing file location from AWS is not straightforward as files are private by default
-        Date expiration = new Date();
-        long expTimeMillis = Instant.now().toEpochMilli();
-        expTimeMillis += SIXY_MINUTES;
-        expiration.setTime(expTimeMillis);
-
-        URL url = s3Client.generatePresignedUrl(bucketName, filename, expiration);
-        return url.toString();
+
+        Duration expiration = Duration.ofMinutes(SIXTY_MINUTES);
+
+        try (s3Presigner) {
+
+            GetObjectRequest getObjectRequest = GetObjectRequest.builder()
+                                                                .bucket(bucketName)
+                                                                .key(filename)
+                                                                .build();
+
+            GetObjectPresignRequest presignRequest = GetObjectPresignRequest.builder()
+                                                                            .getObjectRequest(getObjectRequest)
+                                                                            .signatureDuration(expiration)
+                                                                            .build();
+
+            URL presignedUrl = s3Presigner.presignGetObject(presignRequest).url();
+            return presignedUrl.toString();
+        } catch (Exception e) {
+            LOGGER.error("An exception occurred while presigning a URL", e);
+            return null;
+        }
```

Why did you start swallowing the exception? Given that it wasn't swallowing the exception previously you're changing the behaviour of the software. 🤨

---

noted, I will rethrow it.